web/client/src/components/infrastructure/infrastructureMap.ts

```diff
@@ -0,0 +1,56 @@
+// @ts-nocheck as this is currently unused
+import { iterate } from '@/util/iterate.js';
+import { Map, Source } from 'maplibre-gl';
+
+function highlightPath(mapSource: Source, infrastructure: string, nodeID: string, nodePath: never) {
+	const pathGeoJSON = {
+		type: 'Feature',
+		properties: { id: nodeID },
+		geometry: {
+			coordinates: [],
+			type: 'LineString'
+		}
+	};
+
+	for (const node of iterate(nodePath)) {
+		const gps = infrastructure.element_coords.get(node.element.id);
+		pathGeoJSON.geometry.coordinates.push([gps.lon, gps.lat]);
+	}
+
+	// this is probably illegal, but very convenient
+	const newData = mapSource._data;
+
+	newData.features.push(pathGeoJSON);
+	mapSource.setData(newData);
+}
+
+function deHighlightPath(mapSource: Source, pathID: string) {
+	// again: probably quite unethical
+	const newData = mapSource._data;
+	const idx = newData.features.findIndex(f => f.properties.id === pathID);
+	newData.features.splice(idx, 1);
+
+	mapSource.setData(newData);
+}
+
+export function highlightSignalStationRoute(map: Map, infrastructure: string, signalStationRouteID: string) {
+	highlightPath(map.getSource('signal-station-routes'),
+		infrastructure,
+		signalStationRouteID,
+		infrastructure.signal_station_routes.get(signalStationRouteID).nodes);
```

Comment:
```suggestion
	highlightPath(map.getSource('signal-station-routes'),
		infrastructure,
		signalStationRouteID,
		infrastructure.signal_station_routes.get(signalStationRouteID).nodes,
	);
```

Reply:
I will add a linter Rule for this, as (also in your suggested change) the map.getSource is on the same line as the initial function call.